model_compression_toolkit/data_generation/pytorch/model_info_exctractors.py

```diff
@@ -126,6 +131,27 @@ def __init__(self,
                 hook_handle = module.register_forward_hook(hook.hook)# Register the hook on the module
                 self.hook_handles.append(hook_handle)# Store the hook handle in the hook_handles list
 
+        # This code finds the output layers of the model and adds hooks to the input activation of
+        # those layers. We use torch.fx.symbolic_trace to get a static graph to find the output layers.
+        output_nodes = []
+        self.out_module_weights = []
+
+        # Find the output nodes in the graph
+        for node in fx_model.graph.nodes:
+            if node.op == OUTPUT:
+                output_nodes += node.all_input_nodes
+
+        # Add hooks to the output layers if they are a Linear or Conv2d layer
+        for out in output_nodes:
+            for name, module in model.named_modules():
+                if name == out.target:
+                    if isinstance(module, Linear) or isinstance(module, Conv2d):
```

Comment:
Consider defining Linear/Conv2d as an input to the function. So user can add conv3d for example